### PR TITLE
chore: cleanup agent message model and events

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- remove `agent_messages.content` and `agent_messages.chainOfThoughts` columns (replaced by the raw contents in `AgentMessageContent`)
- "agent_generation_success" and "agent_chain_of_thought" aren't useful for the client / API so we will no longer yield them all the way up ("agent_chain_of_thought" doesn't get out of multi actions loop and "agent_generation_success" doesn't get out of "streamRunAgentEvents")

## Risk

Critical path + migration

## Deploy Plan

1. deploy first migration (39) to allow null values in the chainOfThoughts column
2. deploy code (remove all uses of the `content` and `chainOfThoughts`, stop populating them)
3. apply second migration (40) to drop the 2 columns

### CODE REVIEW

File: front/lib/api/assistant/conversation.ts [hunk 1]
```diff
@@ -1745,9 +1734,7 @@ async function* streamRunAgentEvents(
         runIds.push(event.runId);
         await agentMessageRow.update({
           runIds,
-          content: event.text,
```
Comment (fontanierh) [inline front/lib/api/assistant/conversation.ts:1748]: So right now the only reason we keep `agent_generation_success`  as an event is that we need `runId`. I think we could decide to have the run ID in `AgentMessageContent` event and use that one instead.  cc @spolu

Comment (spolu) [reply front/lib/api/assistant/conversation.ts:1748]: I think it's good to have that event esp if it has a purpose. It makes a more comprehensible API for our users also IMHO.

Comment (fontanierh) [reply front/lib/api/assistant/conversation.ts:1748]: > I think it's good to have that event esp if it has a purpose

But it doesn't have a purpose really. This event is always immediately followed by `agent_message_success`, which contain all agent message info. 
Less events make the API simpler.

Comment (fontanierh) [reply front/lib/api/assistant/conversation.ts:1748]: (With the PR I am proposing this event is **no longer** exposed to end users)

**Review by spolu — APPROVED** [review]
LGTM

We probably want to update our documentation of the events?

**Comment by fontanierh** [comment]
@spolu  I'll take a stab at the docs in the next PR, there's indeed a fair amount to update wrt multi-actions in general

**Pull request merged**